src/bmp.imageio/bmpinput.cpp

```diff
@@ -136,12 +136,12 @@ BmpInput::read_native_scanline(int subimage, int miplevel, int y, int z,
     // if the height is positive scanlines are stored bottom-up
     if (m_dib_header.width >= 0)
         y = m_spec.height - y - 1;
-    const int scanline_off = y * m_padded_scanline_size;
+    const int64_t scanline_off = y * m_padded_scanline_size;
 
-    std::vector<unsigned char> fscanline(m_padded_scanline_size);
+    std::unique_ptr<unsigned char[]> fscanline(new unsigned char[m_padded_scanline_size]);
     fsetpos(m_fd, &m_image_start);
```

Comment:
Shouldn't calls to `fsetpos` also be adjusted? (Even if in practice its likely to be near the start of the file and shouldn't need a 64-bit offset)

Also, wouldn't it be possible to combine the `fsetpos` and `fseek` calls into one?

Reply:
`man fsetpos/fgetpos` says: " These functions provide  a portable way to seek to offsets larger than those that can be represented by a long int." So they are 64 bit safe already. 

I didn't write this code originally, so I was trying to make my changes minimal lest I break something. But I think you're right that they can probably be eliminated in favor of 64 bit safe fseek/ftell. Let me take a look.

Reply:
Yes, replaced and update pushed.